python/ray/__init__.py

```diff
@@ -253,6 +253,23 @@ def __getattr__(self, attr):
 ]
 
 
+# Public APIs that should automatically trigger ray.init().
+AUTO_INIT_APIS = [
+    "cancel",
+    "get",
+    "get_actor",
+    "get_gpu_ids",
+    "get_runtime_context",
+    "kill",
+    "put",
+    "wait",
+]
```

Comment:
How can we collect the full list of all APIs? I checked `@PublicAPI`. There are too many. And most of them are not Ray core APIs. 

> I think it's actually a bug that cluster_resources / available_resources isn't enabled in auto init.

Which bug do you refer to? The bug reported in #34631 is because `get_runtime_context` doesn't auto init. "cluster_resources / available_resources" still don't auto init. Not sure whether they should.

Reply:
To clarify, I mean the previous list of APIs in "__all__", the list of 20 or so things including ObjectID.

Reply:
sounds good. Refactored the "__init__.py" file. do you want to take a another look?